string id
string ip
int32 port
string state